pkg/flowaggregator/flowaggregator_test.go

```diff
@@ -278,3 +286,86 @@ func createElement(name string, enterpriseID uint32) ipfixentities.InfoElementWi
 	ieWithValue, _ := ipfixentities.DecodeAndCreateInfoElementWithValue(element, nil)
 	return ieWithValue
 }
+
+func TestFlowAggregator_watchConfiguration(t *testing.T) {
+	dbEP := &clickhouseclient.ClickHouseExportProcess{}
+	testcases := []struct {
+		FlowCollectorEnable       bool
+		FlowCollectorAddress      string
+		ClickHouseEnable          bool
+		externalFlowCollectorAddr string
+		message                   []updateParam
+		dbExportProcess           *clickhouseclient.ClickHouseExportProcess
+	}{
+		{true, "10.10.10.10:155", false, "", []updateParam{updateExternalFlowCollectorAddr}, nil},
+		{false, "", true, "addr", []updateParam{disableFlowCollector}, nil},
+		{false, "", true, "", []updateParam{enableClickHouse}, nil},
+		{false, "", true, "", []updateParam{updateClickHouseParam}, dbEP},
+		{true, "10.10.10.10:155", false, "", []updateParam{updateExternalFlowCollectorAddr, disableClickHouse}, dbEP},
+	}
+
+	for i, tc := range testcases {
+		t.Run(fmt.Sprintf("subtest: %d", i), func(t *testing.T) {
+			stopCh := make(chan struct{})
+			opt := util.Options{
+				Config: &flowaggregatorconfig.FlowAggregatorConfig{
+					FlowCollector: flowaggregatorconfig.FlowCollectorConfig{
+						Enable:  tc.FlowCollectorEnable,
+						Address: tc.FlowCollectorAddress,
+					},
+					ClickHouse: flowaggregatorconfig.ClickHouseConfig{
+						Enable: tc.ClickHouseEnable,
+					},
+				},
+			}
+			// create watcher
+			fileName := fmt.Sprintf("test_%d.config", i)
+			configWatcher, err := fsnotify.NewWatcher()
+			assert.NoError(t, err)
+			flowAggregator := &flowAggregator{
+				// use a larger buffer to prevent the buffered channel from blocking
+				updateCh:                  make(chan updateMsg, 100),
+				externalFlowCollectorAddr: tc.externalFlowCollectorAddr,
+				dbExportProcess:           tc.dbExportProcess,
+				configFile:                fileName,
+				configWatcher:             configWatcher,
+			}
+			dir := filepath.Dir(fileName)
+			// create default file
+			f, err := os.Create(flowAggregator.configFile)
+			assert.NoError(t, err)
+			// add default file to watcher
+			err = flowAggregator.configWatcher.Add(dir)
+			assert.NoError(t, err)
+			// chmod and delete file, then create a new one (mimic replacement)
```

Comment:
I wouldn't bother with the chmod honestly:
* we don't exactly mimic what happens in K8s anyway, since we don't use a symlink
* we simplified the watch logic and we no longer filter by events

Reply:
You're right. I've deleted chmod.